detox/src/devices/drivers/android/AndroidDriver.js

```diff
@@ -64,52 +63,42 @@ class AndroidDriver extends DeviceDriverBase {
     return await this.aapt.getPackageName(binaryPath);
   }
 
-  async allocateDevice(deviceQuery) {
-    log.debug({ event: ALLOCATE_DEVICE_LOG_EVT }, `Trying to allocate a device based on "${deviceQuery}"`);
-    const adbName = await this.deviceRegistry.allocateDevice(() => this.doAllocateDevice(deviceQuery));
-    log.debug({ event: ALLOCATE_DEVICE_LOG_EVT }, `Settled on ${adbName}`);
-    return adbName;
-  }
-
-  /**
-   * @protected
-   * @return {Promise<string>} adbName of a free matching device
-   */
-  async doAllocateDevice(deviceQuery) {
-    throw Error('Not implemented!');
-  }
-
```

Comment:
Putting this here was a bad idea to begin with. DeviceRegistry is not easy to used generically ATM.
Instead, a more permissive version of this code is now in `AndroidDeviceAllocator`.